plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -522,6 +542,44 @@ func (plugin *InterfaceConfigurator) modifyRxMode(ifIdx uint32, newIntf *intf.In
 	return err
 }
 
+func (plugin *InterfaceConfigurator) modifyContainerIPAddress(newConfig *intf.Interfaces_Interface, oldConfig *intf.Interfaces_Interface, ifIdx uint32) (int8, error) {
+	oldAddr, _, err := addrs.ParseIPWithPrefix(oldConfig.ContainerIpAddress)
+	if err != nil {
+		return 1, err
+	}
+	newAddr, _, err := addrs.ParseIPWithPrefix(newConfig.ContainerIpAddress)
+	if err != nil {
+		return 1, err
+	}
+	toBeDeleted, toBeAdded := addrs.DiffAddr([]*net.IPNet{newAddr}, []*net.IPNet{oldAddr})
```

Comment:
I think this method is too complicated. Why you are calling DiffAddr on two slices where both contain just one item? You can just compare strings:

if oldConfig.ContainerIpAddress != newConfig.ContainerIpAddress {
       // convert to *net.IPNet and make the update
}

Reply:
because I wasn't sure whether in future there won't be more container IP